language/evm/move-to-yul/src/functions.rs

```diff
@@ -107,48 +108,260 @@ impl<'a> FunctionGenerator<'a> {
+                emitln!(ctx.writer, "switch $t{} ", cond);
+                emit!(ctx.writer, "case 0 ");
+                if let StacklessSCG::BasicBlock {
+                    start_offset,
+                    end_offset,
+                } = **if_false
+                {
+                    ctx.emit_block(|| {
+                        for offs in start_offset..end_offset + 1 {
+                            self.bytecode(
+                                ctx,
+                                fun_id,
+                                target,
+                                &label_map,
+                                &code[offs as usize],
+                                false,
+                                true,
+                            );
+                        }
+                    });
+                }
+                emit!(ctx.writer, "default ");
+                if let StacklessSCG::BasicBlock {
+                    start_offset,
+                    end_offset,
+                } = **if_true
+                {
+                    ctx.emit_block(|| {
+                        for offs in start_offset..end_offset + 1 {
+                            self.bytecode(
+                                ctx,
+                                fun_id,
+                                target,
+                                &label_map,
+                                &code[offs as usize],
+                                false,
+                                true,
+                            );
                         }
+                    });
+                }
+            }
+            // TODO: need to emit codes of loops based on StacklessSCG::LoopBlock
+            // based on new bytecodes of Break and Continue
+            StacklessSCG::LoopBlock {
+                loop_header,
+                loop_body,
+            } => {
+                self.emit_scg(&*loop_header, &cfg, code, ctx, fun_id, target);
+                emitln!(ctx.writer, "for {} true {} ");
+                ctx.emit_block(|| {
+                    for loop_body_scg in loop_body {
+                        self.emit_scg(loop_body_scg, &cfg, code, ctx, fun_id, target);
                     }
-                })
+                });
             }
-        });
-        emitln!(ctx.writer)
+        }
+    }
+
+    pub fn push_non_loop_scg(
+        scg_vec: &mut Vec<StacklessSCG>,
+        blk_id: BlockId,
+        cfg: &StacklessControlFlowGraph,
+        visited_blocks: &mut BTreeSet<BlockId>,
+        code: &[Bytecode],
+    ) {
+        let label_map = Self::compute_label_map(cfg, code);
+        let get_block = |l| label_map.get(l).expect("label has corresponding block");
+        if let BlockContent::Basic { lower, upper } = cfg.content(blk_id) {
+            let mut start = *lower;
+            for offs in *lower..*upper + 1 {
+                match &code[offs as usize] {
+                    Bytecode::Branch(_, if_t, if_f, cond) => {
```

Comment:
The branch is always at the end of the block -- this is the standard definition of basic blocks.

That's why we also do not actually need break and continue bytecodes. The SCG data structure simply could have a marker at the end of each block whether it ends with a continue or break.

Reply:
sry that I am not following here, do you mean that here I should try to match other bytecodes like Assign rather than Branch?

> a marker at the end of each block whether it ends with a continue or break
is this the same idea as we chatted last time, to computing a map from label to "block position" and determine if we put a continue or break there?

Thanks for the patience!

Reply:
> sry that I am not following here, do you mean that here I should try to match other bytecodes like Assign rather than Branch?

I mean that you don't need to loop from lower to upper, you just look at the last instruction in the block.

> 
> > a marker at the end of each block whether it ends with a continue or break
> 
> is this the same idea as we chatted last time, to computing a map from label to "block position" and determine if we put a continue or break there?

I mean we do not need new instructions, We can rather represent break and continue like in `SCG::Block(start, end, block_kind)` where `eum BlockKind { Fallthrough, Break, Continue} ` or something like this.

> 
> Thanks for the patience!